static/app/bootstrap/bootstrapRequests.tsx

```diff
@@ -0,0 +1,200 @@
+// XXX(epurkhiser): Ensure the LatestContextStore is initialized before we set
+// the active org. Otherwise we will trigger an action that does nothing
+import 'sentry/stores/latestContextStore';
+
+import {useEffect} from 'react';
+
+import {type ApiResult, Client} from 'sentry/api';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import ProjectsStore from 'sentry/stores/projectsStore';
+import TeamStore from 'sentry/stores/teamStore';
+import type {Organization, Team} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+import parseLinkHeader from 'sentry/utils/parseLinkHeader';
+import {queryOptions, useQuery} from 'sentry/utils/queryClient';
+
+// 30 second stale time
+// Stale time decides if the query should be refetched
+const BOOTSTRAP_QUERY_STALE_TIME = 30 * 1000;
+
+// 10 minute gc time
+// gc time is how long it will stay (even if stale) in the query cache
+const BOOTSTRAP_QUERY_GC_TIME = 10 * 60 * 1000;
```

Comment:
quick note on the comment: `gcTime` only “kicks in” once the query becomes unused / unobserved. As long as there is a `useQuery` usage in a mounted component, it won’t be removed from the cache.

since we call `useBootstrapProjectsQuery ` inside `OrganizationContextProvider`, I would think that we’ll _always_ have a mounted query instance, as this Provider will always be mounted, so using a different `gcTime` won’t do a lot.

Reply:
does it affect the persisters at all? Or do the persisters just save whatever is currently in cache.

Reply:
the persisters are a 1:1 copy of what’s in the memory cache.

Reply:
i guess we might have to check the stale age before loading into the store